pkg/kvstore/kvstore.go

```diff
@@ -44,13 +38,94 @@ type KVClient interface {
 	GetWatcher(key string, timeSleep time.Duration) <-chan []policy.NumericIdentity
 
 	Status() (string, error)
+
+	// Get returns value of key
+	Get(key string) ([]byte, error)
+
+	// GetPrefix returns the first key which matches the prefix
+	GetPrefix(prefix string) ([]byte, error)
+
+	// Set sets value of key
+	Set(key string, value []byte) error
+
+	// Delete deletes a key
+	Delete(key string) error
+
+	// CreateOnly atomically creates a key or fails if it already exists
+	CreateOnly(key string, value []byte, lease bool) error
+
+	// CreateIfExists creates a key with the value only if key condKey exists
+	CreateIfExists(condKey, key string, value []byte, lease bool) error
+
+	// ListPrefix returns a list of keys matching the prefix
+	ListPrefix(prefix string) (KeyValuePairs, error)
```

Comment:
I don't think you can decouple listing the kv pairs and starting the watcher.
For example, in the case of etcd, the watcher must be started with the revision immediately following the range query's response. Here you give no way for the caller to 1) get the revision of the range query, and 2) pass it to the creation of the watcher.

Therefore, I think those must be combined into a single method so that enough context can be preserved to create the watcher properly. See for instance the K8s client API's ListAndWatch.